components/VideoModal.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+  <div class="c-video">
+    <div class="c-video__content" @click="openModal()">
+      <img class="c-video__img" :src="img" />
+      <div class="c-video__overlay">
+        <p class="c-video__overlay-text">{{ $t(text) }}</p>
+        <span role="button" class="c-video__overlay-play">
+          <img src="~/static/images/play.svg"/>
+        </span>
+      </div>
+    </div>
+    <div class="c-video__underlay" v-show="isOpen">
+      <span role="button" class="c-video__close" @click="closeModal()">{{ $t('Close') }}</span>
+      <iframe class="c-video__iframe" :src="videoSrc" width="560" height="315" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  props: [
+    'img',
+    'src',
+    'text',
+  ],
+  data() {
+    return {
+      isOpen: false,
+    };
+  },
+  computed: {
+    videoSrc: function () {
+      return this.isOpen ? `${this.src}?autoplay=1` : this.src;
+    }
+  },
+  methods: {
+    openModal() {
+      this.isOpen = true;
+      this.registerClosingEvents();
+    },
+    registerClosingEvents() {
+      document.addEventListener('keyup', this.shortcutHandler);
+      const underlay = (document.getElementsByClassName('c-video__underlay'))[0];
+      underlay.addEventListener('click', this.closeModal);
+    },
+    deregisterClosingEvents() {
+      document.removeEventListener('keyup', this.shortcutHandler);
+      const underlay = (document.getElementsByClassName('c-video__underlay'))[0];
+      underlay.removeEventListener('click', this.closeModal);
+    },
+    closeModal() {
+      this.isOpen = false;
+      this.deregisterClosingEvents();
+    },
+    shortcutHandler(e) {
+      if (e.keyCode === 27) {
+        this.closeModal();
+      }
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+  @import '~/assets/_variables.scss';
+
+  .c-video {
+    &__underlay {
+      background-color: #333333e8;
```

Comment:
what's this color

Reply:
rgba :) It's a dark partly transparent grey